hummingbot/market/mock_api_order_book_data_source.py

```diff
@@ -0,0 +1,165 @@
+from hummingbot.core.data_type.order_book_tracker_data_source import OrderBookTrackerDataSource
+from hummingbot.core.data_type.order_book_tracker_entry import OrderBookTrackerEntry
+from hummingbot.logger import HummingbotLogger
+from hummingbot.market.huobi.huobi_order_book import HuobiOrderBook
+
+
+class MockAPIOrderBookDataSource(OrderBookTrackerDataSource):
+
+    MESSAGE_TIMEOUT = 30.0
+    PING_TIMEOUT = 10.0
+
+    _maobds_logger: Optional[HummingbotLogger] = None
+
+    @classmethod
+    def logger(cls) -> HummingbotLogger:
+        if cls._maobds_logger is None:
+            cls._maobds_logger = logging.getLogger(__name__)
+        return cls._maobds_logger
+
+    def __init__(self, client: TestClient, symbols: Optional[List[str]] = None):
+        super().__init__()
+        self._client: TestClient = client
+        self._symbols: Optional[List[str]] = symbols
+        self._diff_messages: asyncio.Queue = asyncio.Queue()
+        self._snapshot_messages: asyncio.Queue = asyncio.Queue()
+
+    @classmethod
+    async def get_active_exchange_markets(cls) -> pd.DataFrame:
+        raise NotImplementedError("Symbols are required for mock data source")
+
+    async def get_trading_pairs(self) -> List[str]:
+        if not self._symbols:
+            try:
+                active_markets: pd.DataFrame = await self.get_active_exchange_markets()
+                self._symbols = active_markets.index.tolist()
+            except Exception:
+                self._symbols = []
+                self.logger().network(
+                    f"Error getting active exchange information.",
+                    exc_info=True,
+                    app_warning_msg=f"Error getting active exchange information. Check network connection."
+                )
+        return self._symbols
+
+    @staticmethod
+    async def get_snapshot(client: aiohttp.ClientSession, trading_pair: str) -> Dict[str, Any]:
+        # when type is set to "step0", the default value of "depth" is 150
+        async with client.get("/mockSnapshot") as response:
+            response: aiohttp.ClientResponse = response
+            if response.status != 200:
+                raise IOError(f"Error fetching Huobi market snapshot for {trading_pair}. "
+                              f"HTTP status is {response.status}.")
+            parsed_response = await response.json()
+            return parsed_response
+
+    async def get_tracking_pairs(self) -> Dict[str, OrderBookTrackerEntry]:
+        # Get the currently active markets
+        trading_pairs: List[str] = await self.get_trading_pairs()
+        retval: Dict[str, OrderBookTrackerEntry] = {}
+
+        number_of_pairs: int = len(trading_pairs)
+        for index, trading_pair in enumerate(trading_pairs):
+            try:
+                snapshot: Dict[str, Any] = await self.get_snapshot(self._client, trading_pair)
+                snapshot_msg: OrderBookMessage = HuobiOrderBook.snapshot_message_from_exchange(
```

Comment:
It's better to make this more generic and exchange agnostic. Instead of parsing the message with Huobi specific logic, you should pass in a standard order book snapshot message.